### PR TITLE
Fix error checks during certificatePath reading and parsing in azuread

### PR DESCRIPTION
**Pull request by jegorbunov**
Before this PR certificate-based authentication (in case of a `ActiveDirectoryServicePrincipal` or `ActiveDirectoryApplication` fed auth) would panic (nil pointer dereference at `github.com/microsoft/go-mssqldb@v1.7.2/azuread/configuration.go:228`) because of wrong error checks.

### CODE REVIEW

**Comment by shueybubbles** [comment]
thx for the PR.

Yikes, this is a bad test hole. Can you pls add a test to at least verify the negative case of a missing cert?

File: azuread/configuration_test.go [hunk 1]
```diff
@@ -137,3 +141,66 @@ func TestValidateParameters(t *testing.T) {
...
+		expectedErr string
+	}{
+		{
+			name: "ActiveDirectoryServicePrincipal_cert_not_found",
+			dsn: `sqlserver://someserver.database.windows.net?` +
+				`user id=` + url.QueryEscape("my-app-id@my-tenant-id") + "&" +
+				`fedauth=ActiveDirectoryServicePrincipal` + "&" +
+				`clientcertpath=` + nonExistentCertPath + "&" +
+				`applicationclientid=someguid`,
+			expectedErr: "no such file or directory",
```
Comment (shueybubbles) [inline azuread/configuration_test.go]: are these error messages identical on Windows and Linux?

Comment (jegorbunov) [reply azuread/configuration_test.go]: Rewrote the test to make sure os-agnostic error is checked for the case of "no such file or directory". The second case should be already os-agnostic. If you have ideas to make it better, I would appreciate!

**Review by shueybubbles — APPROVED** [review]

**Comment by shueybubbles** [comment]
i have another PR to prep for v1.8.0 https://github.com/microsoft/go-mssqldb/pull/228
